src/main/java/org/gridsuite/actions/server/ContingencyListService.java

```diff
@@ -342,8 +166,8 @@ ScriptContingencyList createScriptContingencyList(UUID id, ScriptContingencyList
         return fromScriptContingencyListEntity(scriptContingencyListRepository.save(entity));
     }
 
-    Optional<ScriptContingencyList> duplicateScriptContingencyList(UUID sourceListId, UUID id) {
-        return getScriptContingencyList(sourceListId).map(s -> createScriptContingencyList(id, s));
+    Optional<PersistentContingencyList> duplicateScriptContingencyList(UUID sourceListId, UUID id) {
+        return getScriptContingencyList(sourceListId).map(s -> createScriptContingencyList(id, (ScriptContingencyList) s));
```

Comment:
then you have to cast here `(ScriptContingencyList) s` why not returning `Optional<ScriptContingencyList>` then